### PR TITLE
systemd: add minutely option to create timer modal

### PR DESCRIPTION
**Pull request by jelly**
Split off the minutely option https://github.com/cockpit-project/cockpit/pull/17592

---

# Create timer minutely option

The create timer dialog now allows to specify a minutely interval for systemd timers.

### CODE REVIEW

File: pkg/systemd/services/timer-dialog.jsx [hunk 1]
```diff
@@ -273,11 +278,30 @@ const CreateTimerDialogBody = ({ owner }) => {
                                 helperTextInvalid = _("Minute needs to be a number between 0-59");
                             }
 
+                            const sec = repeatPatterns[idx].second;
+                            const validationFailedSecond = !(/^[0-9]+$/.test(sec) && sec <= 59 && sec >= 0);
+
+                            if (submitted && repeat == 'minutely' && validationFailedSecond) {
+                                validated = "error";
+                                helperTextInvalid = _("Second needs to be a number between 0-59");
+                            }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:287]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18072-20221220-142732-4b78ccb7-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 2]
```diff
@@ -273,11 +278,30 @@
...
+                                        {repeat == "minutely" && <>
+                                            <TextInput className='delay-number'
+                                                       id={repeat}
+                                                       value={repeatPatterns[idx].second}
+                                                       onChange={second => {
+                                                           const arr = [...repeatPatterns];
+                                                           arr[idx].second = second;
+                                                           setRepeatPatterns(arr);
+                                                       }}
+                                                       validated={submitted && validationFailedSecond ? "error" : "default"} />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:303]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18072-20221220-142732-4b78ccb7-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 3]
```diff
@@ -273,11 +278,30 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                                validated = "error";
+                                helperTextInvalid = _("Second needs to be a number between 0-59");
+                            }
+
                             return (
                                 <FormGroup label={label} key={item.key}
                                            validated={validated}
                                            helperTextInvalid={helperTextInvalid}>
                                     <Flex className="specific-repeat-group" data-index={idx}>
+                                        {repeat == "minutely" && <>
```
Comment (KKoukiou) [inline pkg/systemd/services/timer-dialog.jsx:294]: You don't need the <> wrapper.

**Review by KKoukiou — DISMISSED** [review]

**Review by KKoukiou — APPROVED** [review]
Was not need to repush :)

File: pkg/systemd/services/timer-dialog.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:287]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18072-20230105-125510-7e33db4b-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 4]
```diff
@@ -273,11 +278,30 @@
...
+                                        {repeat == "minutely" &&
+                                            <TextInput className='delay-number'
+                                                       id={repeat}
+                                                       value={repeatPatterns[idx].second}
+                                                       onChange={second => {
+                                                           const arr = [...repeatPatterns];
+                                                           arr[idx].second = second;
+                                                           setRepeatPatterns(arr);
+                                                       }}
+                                                       validated={submitted && validationFailedSecond ? "error" : "default"} />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:303]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18072-20230105-125510-7e33db4b-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**